pkg/trace/otel.go

```diff
@@ -0,0 +1,195 @@
+package trace
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/getoutreach/gobox/internal/logf"
+	"github.com/getoutreach/gobox/pkg/app"
+	"github.com/getoutreach/gobox/pkg/events"
+	"github.com/getoutreach/gobox/pkg/log"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.10.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type otelTracer struct {
+	Config
+	sync.Once
+	serviceName    string
+	tracerProvider *sdktrace.TracerProvider
+}
+
+// Annotator is a SpanProcessor that adds service-level tags on every span
+type Annotator struct {
+	globalTags GlobalTags
+}
+
+func (a Annotator) OnStart(_ context.Context, s sdktrace.ReadWriteSpan) {
+	setf := func(key string, value interface{}) {
+		s.SetAttributes(attribute.String(key, fmt.Sprintf("%v", value)))
```

Comment:
I have some concerns about the `%v` formatter here.  I think there are cases where the input contains one or more nested maps.  

See https://github.com/getoutreach/gobox/blob/v1.40.0/pkg/log/logtest/logtest.go#L70 and https://github.com/getoutreach/gobox/blob/c8a5b56788ef7a8ad27373c63febf6f993e1a1b0/internal/logf/logf.go#L37 for some tools that might help to convert maps to something dot-separated.

Reply:
I'll grant that this seems to be equally unhandled in the Honeycomb implementation.

I don't know if that means this has been broken forever, or if I'm currently misinterpreting how it works.